R/zchunk_LA100.IEA_downscale_ctry.R

```diff
@@ -0,0 +1,308 @@
+             OPTIONAL_FILE = "energy/en_OECD",
+             OPTIONAL_FILE = "energy/en_nonOECD",
+             OPTIONAL_FILE = "energy/L100.IEA_en_bal_ctry_hist",
+             FILE = "energy/mappings/IEA_product_downscaling",
+             FILE = "energy/mappings/IEA_ctry"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L100.IEA_en_bal_ctry_hist"))
+  } else if(command == driver.MAKE) {
+
+    `1990` <- `1990_share` <- `1990_sum` <- COUNTRY <- EcYield_kgm2_hi <- EcYield_kgm2_lo <- FLOW <-
+      GCAM_commodity <- GCAM_region_ID <- GLU <- IEAcomposite <- Irr_Rfd <- LC_bm2_hi <-
+      LC_bm2_lo <- PRODUCT <- composite_population <- iso <- landshare_hi <- landshare_lo <-
+      level <- optional <- value <- year <- yield <- yieldmult_hi <- yieldmult_lo <- NULL # silence package check notes
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    L100.Pop_thous_ctry_Yh <- get_data(all_data, "L100.Pop_thous_ctry_Yh")
+    en_OECD <- get_data(all_data, "energy/en_OECD")
+    en_nonOECD <- get_data(all_data, "energy/en_nonOECD")
+    IEA_product_downscaling <- get_data(all_data, "energy/mappings/IEA_product_downscaling")
+    IEA_ctry <- get_data(all_data, "energy/mappings/IEA_ctry")
+    L100.IEA_en_bal_ctry_hist_preprocessed <- get_data(all_data, "energy/L100.IEA_en_bal_ctry_hist")
+
+    # If the (proprietary) raw IEA datasets are available, go through the full computations below
+    # If not, use the pre-saved summary file (i.e., the output of this chunk!) assuming it's available
+    if(!is.na(en_nonOECD) && !is.na(en_OECD)) {
+
+      # Full calculation from raw data
+      # The two IEA datasets are LARGE and we keep them in wide format for most of the computations
+      # below, as they are very expensive to reshape. As a result, much of the logic below closely
+      # parallels that in the original data system, and there are fewer dplyr-style pipelines.
+
+      # Subset only the relevant years and combine OECD with non-OECD
+      hy <- intersect(intersect(HISTORICAL_YEARS, colnames(en_OECD)), colnames(en_nonOECD))
+      cols <- c("COUNTRY", "FLOW", "PRODUCT", hy)
+      bind_rows(en_OECD[cols], en_nonOECD[cols]) %>%
+        # rename fuels with inconsistent naming between the two databases
+        mutate(PRODUCT = replace(PRODUCT,
+                                 PRODUCT %in% c("Natural Gas", "Other Kerosene", "Total"),
+                                 c("Natural gas", "Other kerosene", "Total of all energy sources"))) ->
+        L100.IEAfull
+
+      # UP FRONT ADJUSTMENTS (UFA) original lines 42-67
+
+      # UFA1. Nearly the entire supply of natural gas in other Africa between 2001 and 2004 is allocated
+      # to GTL plants operating at nearly 100% efficiency. We adjust the energy input quantities
+      # to avoid negative values later on.
+      GTL_COEF <- 1.7
+      GTL_ADJ_YEARS <- as.character(intersect(2001:2004, HISTORICAL_YEARS))
+      if(length(GTL_ADJ_YEARS)) {
```

Comment:
Do we encourage using integer expressions in conditional tests (as opposed to, say `length(GTL_ADJ_YEARS)  > 0`)?  (I don't care one way or the other, but I gather it's considered bad style in higher level languages).